PariStyle/src/main/kotlin/domain/usecase/GetClothingRecommendationUseCase.kt

```diff
@@ -1,9 +1,91 @@
+            clothingItem.suitableConditions.contains(weatherCondition) &&
+                    checkTemperatureRange(temperature,clothingItem)
+        }
+        return suitableClothes.firstOrNull() ?: throw PariStyleException.SuitableClothesException(
+            "No Suitable Clothes available"
+        )
+    }
+
+    private fun getListOfClothesItemsWithDifferentConditions(): List<ClothingItem>{
+        return listOf(
+            ClothingItem(
+                clothingType = ClothingType.T_SHIRT,
+                description = "Light short-sleeve shirt",
+                suitableTemperatureRange = TemperatureRange(20.0, 40.0),
+                suitableConditions = setOf(
+                    WeatherCondition.CLEAR_SKY,
+                    WeatherCondition.MAINLY_CLEAR,
+                    WeatherCondition.PARTLY_CLOUDY
+                )
+            ),
+            ClothingItem(
+                clothingType = ClothingType.SHIRT,
+                description = "Regular long-sleeve shirt",
+                suitableTemperatureRange = TemperatureRange(15.0, 25.0),
+                suitableConditions = setOf(
+                    WeatherCondition.CLEAR_SKY,
+                    WeatherCondition.MAINLY_CLEAR,
+                    WeatherCondition.PARTLY_CLOUDY,
+                    WeatherCondition.OVERCAST
+                )
+            ),
+            ClothingItem(
+                clothingType = ClothingType.RAINCOAT,
+                description = "Waterproof raincoat",
+                suitableTemperatureRange = TemperatureRange(-10.0, 25.0),
+                suitableConditions = setOf(
+                    WeatherCondition.DRIZZLE,
+                    WeatherCondition.LIGHT_FREEZING_DRIZZLE,
+                    WeatherCondition.SLIGHT_RAIN,
+                    WeatherCondition.MODERATE_RAIN,
+                    WeatherCondition.HEAVY_INTENSITY_RAIN,
+                    WeatherCondition.LIGHT_FREEZING_RAIN,
+                    WeatherCondition.HEAVY_INTENSITY_FREEZING_RAIN,
+                    WeatherCondition.SLIGHT_RAIN_SHOWERS,
+                    WeatherCondition.MODERATE_RAIN_SHOWERS,
+                    WeatherCondition.VIOLENT_RAIN_SHOWERS
+                )
+            ),
+            ClothingItem(
+                clothingType = ClothingType.UMBRELLA,
+                description = "Foldable umbrella",
+                suitableTemperatureRange = TemperatureRange(-5.0, 30.0),
+                suitableConditions = setOf(
+                    WeatherCondition.DRIZZLE,
+                    WeatherCondition.SLIGHT_RAIN,
+                    WeatherCondition.MODERATE_RAIN,
+                    WeatherCondition.SLIGHT_RAIN_SHOWERS,
+                    WeatherCondition.MODERATE_RAIN_SHOWERS
+                )
+            )
+        )
+
+    }
+
+    private fun checkTemperatureRange(temperature : Double, clothingItem: ClothingItem): Boolean{
```

Comment:
this for more functional approach 
```suggestion
       private fun isWithinTemperatureRange(temperature: Double, clothingItem: ClothingItem) =
        temperature in clothingItem.suitableTemperatureRange.min..clothingItem.suitableTemperatureRange.max
```